File: backend/api/schemas/qa.py
from pydantic import BaseModel
from typing import Optional, List


class QARequest(BaseModel):
    query: str
    kb_id: Optional[str] = None
    config_name: Optional[str] = "base"


class QAResponse(BaseModel):
    query: str
    final_answer: str
    step_by_step_analysis: str
    reasoning_summary: str
    relevant_pages: List[int]
    used_parent_chunks: List[str]


class FeedbackRequest(BaseModel):
    query: str
    answer: str
    feedback: str            # "good" / "bad"
    comment: Optional[str] = None
    kb_id: Optional[str] = None
    config_name: Optional[str] = "base"
    # 详细差评字段
    error_type: Optional[str] = None      # hallucination / irrelevant / incomplete / factual_error / outdated / other
    correct_answer: Optional[str] = None  # 用户纠正的正确答案
    pages: Optional[List[int]] = None     # 引用页码（从前端传回）

File: frontend/src/api/qa.ts
import http from './http'

export interface QAResponse {
  query: string; final_answer: string; step_by_step_analysis: string
  reasoning_summary: string; relevant_pages: number[]; used_parent_chunks: string[]
}

export interface FeedbackData {
  query: string
  answer: string
  feedback: string        // "good" | "bad"
  comment?: string
  kb_id?: string
  config_name?: string
  pages?: number[]
  error_type?: string
  correct_answer?: string
}

export const qaApi = {
  ask: (query: string, kbId?: string) =>
    http.post<QAResponse>('/qa/ask', { query, kb_id: kbId }),

  feedback: (data: FeedbackData) =>
    http.post('/qa/feedback', data),

  clearHistory: (kbId?: string) =>
    http.delete('/qa/history', { params: { kb_id: kbId } }),
}

/** 创建 SSE 流式问答连接，返回 EventSource */
export function createQAStream(query: string, kbId?: string, configName = 'base'): EventSource {
  const token = localStorage.getItem('token') ?? ''
  const params = new URLSearchParams({ q: query, config_name: configName })
  if (kbId) params.append('kb_id', kbId)
  params.append('token', token)
  return new EventSource(`/api/qa/stream?${params.toString()}`)
}
